src/config_wrapper.cpp

```diff
@@ -144,6 +167,76 @@ void config_wrapper::set_fee_parameters(const fee_parameters& fee_params,
     set_dirty();
 }
 
+void config_wrapper::update_gas_params(eosio::asset ram_price_mb, uint64_t minimum_gas_price) {
+
+    eosio::check(ram_price_mb.symbol == token_symbol, "invalid price symbol");
+    eosio::check(minimum_gas_price >= 1000000000ull, "gas_price must >= 1Gwei");
+
+    double gas_per_byte_f = (ram_price_mb.amount / 10000.0 * 1e18 / (1024.0 * 1024.0)) / 
+        (minimum_gas_price * (double)(100000 - _cached_config.miner_cut) / 100000.0);
```

Comment:
```cpp
double gas_per_byte_f = (ram_price_mb.amount / (1024.0 * 1024.0) * 1e18) / (minimum_gas_price * static_cast<double>(hundred_percent - _cached_config.miner_cut) / hundred_percent);
```

Reply:
Even better if the 1e14 was a `constexpr double` constant in `types.hpp` that was `static_assert`ed to be compatible with the 10 to the power of `evm_precision - token_symbol.precision()`.